### DISCUSSION TITLE
IP allow list instead of blocklist

### DISCUSSION THREAD

**Discussion by RedyAu** [rejetto/hfs#242] [Q&A] [answered]
To save myself some headaches, I'm using Cloudflare in front of HFS. I'd like to limit HFS to only serve Cloudflare's IP ranges (available here: https://www.cloudflare.com/ips/
HFS3's admin panel seemingly only has a block feature. How do I achieve this?

**Comment by RedyAu** [comment]
To provide more context: I have ports open other than 80 on my server, which don't need proxying, for other stuff. However, I want HFS behind a proxy, and unresponsive to direct requests.

**Comment by rejetto** [comment]
hi, you want whitelist and we have only blacklist, so basically I'd say you need the whitelist as a feature (not available yet) or as a plugin.
But is somewhat already possible because we support wildcards and negate operator (!) https://github.com/rejetto/hfs/wiki/Wildcards
(I should link it in-place for that field as well)
For example I just tested this text on my local server and it worked: `!192.*|127.*|::1`
In this way we are turning the blacklist into a whitelist, specifying 3 entries, but the trick works only on a single line, that's why i'm glueing all entries with the `|` character.
But there's still a problem for you: cloudflare use the CIDR syntax for their list, and operators (! and |) are not supported that way.

So your options now are
- write a plugin
- use the syntax I just said to get quite close to the actual list
- wait for me to to do something about it

**Reply by RedyAu** [reply]
If you'd say this is on the roadmap for some point in the future, then I'm in no hurry ;) Thank you!
So far loving this new HFS. HFS2 was great, but it was hard to trust it, was getting a bit old :D

**Comment by rejetto** [comment, marked as answer]
ok, I made it, check this https://github.com/rejetto/hfs/discussions/241

i introduced operators for the CIDR syntax

**Reply by RedyAu** [reply]
Wow, awesome, thanks!
I haven't seen the UI yet, so this may already be in place; in an info box it would be great to have a short version of the wiki page, explaining in a couple different examples what's possible with the blacklist.

**Reply by rejetto** [reply]
that's actually so
